app/controllers/comments_controller.rb

```diff
@@ -1,28 +1,50 @@
 # frozen_string_literal: true
 
 class CommentsController < AuthenticationController
+  before_action :set_comment_type, only: %i[create update]
+
   def create
-    new_comment = policy.comments.new(comment_params)
-    new_comment = policy.comments.new if new_comment.save
+    new_comment = @comment_type.comments.new(comment_params)
+    new_comment = @comment_type.comments.new if new_comment.save
     render(json: { partial: create_comment_partial(new_comment) })
   end
 
   def update
-    comment = policy.comments.find(params[:id])
+    comment = @comment_type.comments.find(params[:id])
     comment.update(deleted_at: DateTime.current)
     render(json: { partial: update_comment_partial })
   end
 
   private
 
   def create_comment_partial(new_comment)
+    if @comment_type.is_a? Policy
+      create_policy_comment_partial(new_comment)
+    else
+      create_evidence_group_comment_partial(new_comment)
+    end
+  end
+
+  def create_policy_comment_partial(new_comment)
     render_to_string(
       partial: "planning_application/review_policy_classes/comment",
       locals: {
         planning_application:,
         policy_class:,
         policy:,
-        comment: policy.comment,
+        comment: @comment_type.comment,
+        new_comment:
+      }
+    )
+  end
+
+  def create_evidence_group_comment_partial(new_comment)
```

Comment:
It's probably not a bug introduced by this PR, but when i'm editing a comment (i.e. creating a new comment), the new comment doesn't immediately appear in the previous comments list so it's hard to know whether the update was successful or not. I have to refresh the page to see the new records. I'm not sure if this is because the collection is being memoized or that the rendering json only updates the comment partial view rather than the previous comments too. We probs want to pass it in the updated list so we can see the update straight away.

Reply:
Yes I noticed this and then was lazy and said that's how it works currently so I don't want to fix it 😂 
I will now try and fix it 😬

Reply:
Ah ok well maybe it is out of scope for this PR then.
Let's add a bug ticket for it perhaps as it might not be so straightforward and shouldn't be a blocker

Reply:
It looks like there aren't that many dev tickets coming up, so may have time before everything's UAT-ed to have a look for this PR

Reply:
Having done an initial think/chat I think this is way too complicated for any gain we might get for now